packages/ic-asset-certification/src/asset_router.rs

```diff
@@ -944,7 +947,9 @@ impl<'content> AssetRouter<'content> {
         let cel_expr_str = cel_expr.to_string();
         headers.push((CERTIFICATE_EXPRESSION_HEADER_NAME.to_string(), cel_expr_str));
 
-        let request = HttpRequest::get(url).build();
+        let request = HttpRequest::get(url)
+            .with_headers(certified_request_headers.clone())
```

Comment:
and then include the full header name/value pair in the request for calculating the request hash